src/condor_tools/store_cred_main.cpp

```diff
@@ -811,6 +776,51 @@ usage()
 	exit( 1 );
 };
 
+bool getCredData(StoreCredOptions& options, char **cred, size_t* credlen)
+{
+	*cred = nullptr;
+	*credlen = 0;
+	int cred_type = (options.mode & CRED_TYPE_MASK);
+
+	if (options.credential_file) {
+		if (MATCH == strcmp(options.credential_file, "-")) {
+			int max_len = 1024 * 1024; // max read from stdin is 1 Mb
+			*cred = (char*)malloc(max_len+1);
+			#ifdef _WIN32
+			// disable CR+LF munging of the stdin stream, we want to treat it as binary data
+			_setmode(_fileno(stdin), _O_BINARY);
+			#endif
+			*credlen = full_read(fileno(stdin), *cred, max_len);
+			if (((ssize_t)credlen) < 0) {
+				fprintf(stderr, "ERROR: could read from stdin: %s\n", strerror(errno));
```

Comment:
credlen is a pointer here,  shouldn't you be comparing the pointed to value in the if statement?